src/app/containers/InlineLink/index.jsx

```diff
@@ -7,29 +7,38 @@ import fragment from '../Fragment';
 import { inlineLinkModelPropTypes } from '../../models/propTypes/inlineLink';
 import { articleRegexPath } from '../../routes';
 
-const InternalInlineLink = InlineLink.withComponent(Link);
-
 const componentsToRender = { fragment };
 
-const InlineLinkContainer = ({ locator, blocks }) => {
-  const regexp = pathToRegexp(articleRegexPath, [], {
+const reactRouterLink = (path, blocks) => {
+  const ReactRouterLink = InlineLink.withComponent(Link);
+
+  return (
+    <ReactRouterLink to={path}>
+      <Blocks blocks={blocks} componentsToRender={componentsToRender} />
+    </ReactRouterLink>
+  );
+};
+
+const pathIfMatching = (regexString, url) => {
+  const regexp = pathToRegexp(regexString, [], {
     start: false,
     end: false,
   });
+  const regexResultArray = regexp.exec(url);
+  if (!regexResultArray) {
+    return null;
+  }
+  const path = regexResultArray[0];
+  return path;
+};
+
+const InlineLinkContainer = ({ locator, blocks }) => {
+  const path = pathIfMatching(articleRegexPath, locator);
 
-  const result = regexp.exec(locator);
-  // if URL matches a valid route, use a react-router link
-  if (result) {
-    // the path is the first item in the array
-    const path = result[0];
-    return (
-      <InternalInlineLink to={path}>
-        <Blocks blocks={blocks} componentsToRender={componentsToRender} />
-      </InternalInlineLink>
-    );
+  if (process.env.TOGGLE_ENABLE_CLIENTSIDE_ROUTING === 'true' && path) {
```

Comment:
I've been mulling this over since I first looked at this... Comparing this to other implementations I've seen, I think we _might_ want a central location where we can see all extant feature flags? 

It'd maybe look a bit unnecessary to fresh eyes, but we could have a single `FeatureToggle.js` which contains constants like:
`export const FF_ENABLE_CLIENTSIDE_ROUTING = process.env.ENABLE_CLIENTSIDE_ROUTING`

I'm aware this might sound really trivial, but when our flags are scattered throughout our codebase, & someone inevitably doesn't follow naming schemes/makes a typo, it might be beneficial. We need to be able to add & remove flags quickly & painlessly.